drivers/sx127x/sx127x_netdev.c

```diff
@@ -0,0 +1,499 @@
+                        packet_info->rssi = SX127X_RSSI_OFFSET_HF + rssi + (rssi >> 4) + packet_info->snr;
+                    }
+                    else {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_LF + rssi + (rssi >> 4) + packet_info->snr;
+                    }
+#endif
+                }
+                else {
+#if defined(MODULE_SX1272)
+                    packet_info->rssi = SX127X_RSSI_OFFSET + rssi + (rssi >> 4);
+#else /* MODULE_SX1276 */
+                    if (dev->settings.channel > SX127X_RF_MID_BAND_THRESH) {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_HF + rssi + (rssi >> 4);
+                    }
+                    else {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_LF + rssi + (rssi >> 4);
+                    }
+#endif
+                }
+            }
+
+            packet_info->time_on_air = sx127x_get_time_on_air(dev);
+
+            size = sx127x_reg_read(dev, SX127X_REG_LR_RXNBBYTES);
+            if (buf == NULL) {
+                return size;
+            }
+
+            if (size > len) {
+                return -ENOBUFS;
+            }
+
+            if (!dev->settings.lora.rx_continuous) {
+                sx127x_set_state(dev, SX127X_RF_IDLE);
+            }
+
+            xtimer_remove(&dev->_internal.rx_timeout_timer);
+
+            /* Read the last packet from FIFO */
+            uint8_t last_rx_addr = sx127x_reg_read(dev, SX127X_REG_LR_FIFORXCURRENTADDR);
+            sx127x_reg_write(dev, SX127X_REG_LR_FIFOADDRPTR, last_rx_addr);
+            sx127x_read_fifo(dev, (uint8_t*)buf, size);
+            break;
+        default:
+            break;
+    }
+
+    return size;
+}
+
+static int _init(netdev_t *netdev)
+{
+    sx127x_t *sx127x = (sx127x_t*) netdev;
+
+    sx127x->irq = 0;
+    sx127x_radio_settings_t settings;
+    settings.channel = SX127X_CHANNEL_DEFAULT;
+    settings.modem = SX127X_MODEM_DEFAULT;
+    settings.state = SX127X_RF_IDLE;
+
+    sx127x->settings = settings;
+
+    /* Launch initialization of driver and device */
+    DEBUG("init_radio: initializing driver...\n");
+    sx127x_init(sx127x);
+
+    sx127x_init_radio_settings(sx127x);
+    /* Put chip into sleep */
+    sx127x_set_sleep(sx127x);
+
+    DEBUG("init_radio: sx127x initialization done\n");
+
+    return 0;
+}
+
+static void _isr(netdev_t *netdev)
+{
+    sx127x_t *dev = (sx127x_t *) netdev;
+
+    uint8_t irq = dev->irq;
+    dev->irq = 0;
+
+    switch (irq) {
+    case SX127X_IRQ_DIO0:
+        sx127x_on_dio0(dev);
+        break;
+    case SX127X_IRQ_DIO1:
+        sx127x_on_dio1(dev);
+        break;
+    case SX127X_IRQ_DIO2:
+        sx127x_on_dio2(dev);
+        break;
+    case SX127X_IRQ_DIO3:
+        sx127x_on_dio3(dev);
+        break;
+    case SX127X_IRQ_DIO4:
+        sx127x_on_dio4(dev);
+        break;
+    case SX127X_IRQ_DIO5:
+        sx127x_on_dio5(dev);
+        break;
+    default:
+        break;
```

Comment:
according to RIOT coding style, that points to [Linux kernel coding style guidelines](https://www.kernel.org/doc/Documentation/process/coding-style.rst), this one is correct, the others aren't.